build error with master and node v0.8 and debian

It looks like the `'type': 'executable'` is being ignored with older node/node-gyp versions such that a `topcube.node` is being created instead, which is incorrect. I will try to replicate and then provide a pull request fixing.

Downstream issue: http://support.mapbox.com/discussions/tilemill/6916-npm-errors-when-trying-to-install-tilemill-from-official-mapbox-docs

```
Release/topcube.node: Finished make: *** No rule to make targetRelease/topcube', needed by bin/topcube'. Stop. make: Leaving directory/home/peter/node_modules/tilemill/node_modules/topcube/build'
```

```
gyp ERR! node -v v0.8.9
gyp ERR! node-gyp -v v0.6.8
```

--- a/package.json
+++ b/package.json
@@ -10,5 +10,5 @@
   "main": "./lib/topcube.js",
   "dependencies": {},
   "devDependencies": {},
-  "engines": { "node": "0.6.x || 0.8.x || 0.10.x" }
+  "engines": { "node": ">= 0.8.11 < 0.11.0" }
 }

